src/time-resolution.jl

```diff
@@ -65,19 +68,31 @@ function compute_constraints_partitions(graph, representative_periods)
         ),
     ]
 
-    num_rep_periods = length(representative_periods)
+    RP = Dict{Int,UnitRange{Int}}(year => 1:length(representative_periods[year]) for year in years)
+
+    active_years = Dict{String,Vector{Int}}(
+        a => [year for year in years if graph[a].active[year]] for
+        a in MetaGraphsNext.labels(graph)
+    )
 
     for (name, partitions, strategy, asset_filter) in partitions_cases
         constraints_partitions[name] = OrderedDict(
-            (a, rp) => begin
-                P = partitions(a, rp)
+            (a, iy, y, rp) => begin
+                P = partitions(a, iy, y, rp)
                 if length(P) > 0
-                    compute_rp_partition(partitions(a, rp), strategy)
+                    result = compute_rp_partition(partitions(a, iy, y, rp), strategy)
+                    result
                 else
                     Vector{TimestepsBlock}[]
                 end
-            end for
-            a in MetaGraphsNext.labels(graph), rp in 1:num_rep_periods if asset_filter(a)
+            end for a in MetaGraphsNext.labels(graph) if asset_filter(a) for
+            y in active_years[a] for iy in (
+                if any(graph[a].investable[iy] for iy in years)
+                    filter(iy -> graph[a].investable[iy], years)
+                else
+                    [0]
```

Comment:
```suggestion
                    [false]
```
Trying to translate.
If the graph is investable at any of the years, then loop it over the values of `investable` for each year.
Else, loop `yi` over `[0]`.
This is a problem because you are returning a vector of booleans in the first and a vector of integers in the second.

You can also try:

```suggestion
                    Bool[]
```

Unless you need the single value there.

Also, just try `for it in filter(iy -> graph[a].investable[iy], years)`. I think maybe it won't work (and you might have tried already).

Reply:
Actually I want to return a list of investment years, for which `iy` (investment year) can be looped over. So I'm trying to say, if the asset is investable at any of the years, then get the list of years for which the `graph[a].investable[iy]` is true.
Else, use a default list of [0], i.e., year 0.